exonum-java-binding/core/src/main/java/com/exonum/binding/util/LibraryLoader.java

```diff
@@ -20,29 +20,81 @@
 import org.apache.logging.log4j.Logger;
 
 /**
- * A loader of the native shared library with Exonum framework bindings.
+ * A loader of the native shared library with Exonum framework bindings. It loads the native
+ * library and also verifies that it is compatible with the Java classes. The native library
+ * is compatible iff it has exactly the same version as this Java library. The revision
+ * from which they were built is currently not checked, but may be in the future (see ECR-3173),
+ * because the API between Java and native is considered internal and can be changed
+ * in an incompatible way in any revision.
  *
- * <p>To have library java_bindings available by its name,
- * add a path to the folder containing it to <code>java.library.path</code> property,
- * e.g.: <code>java -Djava.library.path=${EXONUM_HOME}/lib/native …</code>
+ * <p>To enable loading of java_bindings library, add a path to the folder containing it
+ * to <code>java.library.path</code> property, e.g.:
+ * <code>java -Djava.library.path=${EXONUM_HOME}/lib/native …</code>
+ *
+ * <p>This class is thread-safe.
+ *
+ * @see <a href="https://exonum.com/doc/version/0.11/get-started/java-binding/#installation">
+ *   Exonum Java installation instructions</a>
+ * @see <a href="https://exonum.com/doc/version/0.11/get-started/java-binding/#testing">
+ *   Build configuration to enable integration testing of Java services</a>
  */
 public final class LibraryLoader {
 
   private static final String BINDING_LIB_NAME = "java_bindings";
   private static final String JAVA_LIBRARY_PATH_PROPERTY = "java.library.path";
   private static final String DYNAMIC_LIBRARIES_ENV_VAR_WINDOWS = "PATH";
   private static final String DYNAMIC_LIBRARIES_ENV_VAR_UNIX = "LD_LIBRARY_PATH";
+  // TODO: Shall this class read the properties instead?
```

Comment:
We could make Maven auto-generate the properties file with the version, but … not sure we need that at this moment (with some code here reading those properties).

Reply:
Yes — update this string each release or make Maven write and this code read the build properties.

Reply:
Would it be possible/better to generate source file with constants before building the "static" source code? (move it to compilation phase instead of reading properties at runtime)

Reply:
It's trivial to pick up maven properties in runtime. So I believe it brings more benefits than drawbacks 🦊 
Also, looks like we can configure RUSTFLAGS in maven and remove `tests_profile`? 🤔

Reply:
> It's trivial to pick up maven properties in runtime. So I believe it brings more benefits than drawbacks fox_face

Yes, it is not _difficult_ at all, but it seems like an overkill at this point: this class will gain a responsibility to read (once) a certain file with properties + the build configuration needs to be updated.

Reply:
> Also, looks like we can configure RUSTFLAGS in maven and remove tests_profile

How? It invokes rustup and find do determine some various paths to add to RUSTFLAGS.

Reply:
For example:
```
    <plugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-antrun-plugin</artifactId>
        <version>1.7</version>
        <executions>
          <execution>
            <id>define-property</id>
            <phase>validate</phase>
            <goals>
              <goal>run</goal>
            </goals>
            <configuration>
              <exportAntProperties>true</exportAntProperties>
              <tasks>
                <exec executable="git" outputproperty="testproperty">
                  <arg value="--version"/>
                </exec>
              </tasks>
            </configuration>
          </execution>
          <execution>
            <id>show-property</id>
            <phase>compile</phase>
            <goals>
              <goal>run</goal>
            </goals>
            <configuration>
              <tasks>
                <echo>Displaying value of 'testproperty':</echo>
                <echo>[testproperty] ${testproperty}</echo>
              </tasks>
            </configuration>
          </execution>
        </executions>
      </plugin>
    </plugins>
```
It'll print in the log:
```
     [echo] Displaying value of 'testproperty':
     [echo] [testproperty] git version 2.20.1 (Apple Git-117)
```

Reply:
Looks interesting, maybe we can offload that to an Ant script instead of a bash script for ease of use (no source tests_profile) and support of any platform (Windows). Please file an issue if consider that promising.

Reply:
ECR-3184